### ISSUE TITLE
Add Ollama capability to gpt/ask_gpt.py

### ISSUE THREAD

**Issue by thimbleforth** [cmu-sei/dmc#1]
**TLDR:** GPT4 and `ollama` use the same API structure. Adding [ollama](https://github.com/ollama/ollama) to `dmc` would significantly increase the diversity of available models for integration into the software's capability, with a minimum of effort.

**Summary:** Adding functionality for `dmc` to be able to use `ollama` would allow for self-contained, self-hosted deployment of LLM-assisted static flow analysis. This could also be useful in environments where external LLM services are undesirable, unaffordable, or otherwise blocked by policy.

**How**: 
* Refactor `gpt/ask_gpt.py` to leverage the [ollama-python library](https://github.com/ollama/ollama-python), in addition to the OpenAI library.
* Require inputs at runtime, whether through a given file or terminal input, which determine OpenAI or Ollama are used, what model, &etc.
* Create two different branches of the code flow which will create the given API client on demand for whichever LLM service is chosen
* Remove hardcoded messages definitions and GPT4 references within `ask_func()`

Since I'm suggesting these changes I'm happy to work through them and send in a PR.
